### ISSUE TITLE
Separate settings for running tests

### ISSUE THREAD

**Issue by cameel** [golemfactory/concent#28]
Currently we're always running automated unit and integration tests with programmer's `local_settings.py` which introduces some variability to the environment. Tests often assume specific settings and can fail if they're different.

To some extent this is solved by using `override_settings` decorator every time a test depends on a specific value but this has several issues:
- If there are a lot of settings to override this is overly verbose
- It may not be apparent to the programmer that the code under test depends on a specific value, especially when it comes to built-in Django settings
- Some settings must be always overridden to prevent destroying data. For example addresses of external services or databases. They should be mocked during tests. If they're not and the address from real settings is accidentally used by the test, the code may connect to the real service. With `override_settings` it's easy to miss it in some cases.

For these reasons we want to have a separate settings file just for testing.
- Create `testing.py` in `concent_api/concent_api/settings/`, analogous to `production.py` and `development.py`
    - `testing.py` should import `base.py`.
    - Our custom settings should be undefined whenever possible. Define them only if not doing it breaks system check or would require using `override_settings` on nearly everything. Preferably set it to `None`.
        - Most importantly, all URLs and backend addresses should not point at real services. Best if they can be left blank and overriden only for specific tests that need a value.
    - Most Django settings should be left at values from `base.py`.
        - `DATABASE['NAME']` should be not set and `DATABASE['TEST']['NAME']` hard-coded to `test_concent_api`.

          ``` python
          DATABASES = {
              'default': {
                  # NAME intentionally left unset, in case tests try to access the production database.
                  # This could happen e.g. if someone executes a query at module level.
                  'ENGINE':          'django.db.backends.postgresql_psycopg2',
                  'USER':            'postgres',
                  'ATOMIC_REQUESTS': True,
                  'TEST': {
                      'NAME': 'test_concent_api'
                  }
              }
          }
          ```
    - Normal logging should be disabled during tests

      ``` python
      LOGGING = {
          'version': 1,
          'disable_existing_loggers': True,
          'handlers': {
              'null_handler': {'class': 'logging.NullHandler'}
          },
          'loggers': {
              '': {'handlers': ['null_handler']}
          }
      }
      ```
- Adjust `full_check.sh` to run `manage.py` with the newly created settings (`--settings` command-line option)

**Comment by rwrzesien** [comment]
@cameel Tests are always creating new empty database for each run and base with name from settings is not touched at all:
https://docs.djangoproject.com/en/2.0/topics/testing/overview/#the-test-database
The config entry which you have added will do exactly the same.

**Comment by cameel** [comment]
Yes it does the same thing but only if you have `DATABASE['NAME']` set because that's what Django uses to figure out the name for the test database. And I want `DATABASE['NAME']` to be missing in this config to prevent anything from accessing the real database in any circumstances.

This is just in case. Normally tests will not touch the normal database but nothing is actually preventing them from doing so. It's possible to write code that so broken that the code starts writing to the normal database during tests. This happens if the code runs before tests settings are loaded. For example at module level. We've run into a situation like that in an earlier project. It's a pretty rare case and I wouldn't go out of my way to safeguard against it but since we're doing the separate config anyway, it does not cost us much to make the name of the normal database blank which makes this case completely impossible.

A better way to secure the real database is actually to use a role with limited privileges for running tests and we're going to do that to. But it's trivial to do this sagefuard here to do why not do it.

**Comment by rwrzesien** [comment]
Ok, I will add it then.

**Comment by rwrzesien** [comment]
@cameel Pull request created.

**Comment by cameel** [comment]
Done in #30 and merged into master: ab67cdbfdc06990e2463d7c20beb083d6af2235d
